internal/services/hdinsight/hdinsight_hadoop_cluster_resource.go

```diff
@@ -258,6 +260,20 @@ func resourceHDInsightHadoopClusterCreate(d *pluginsdk.ResourceData, meta interf
 		Tags:     tags.Expand(t),
 		Identity: identity,
 	}
+
+	if v, ok := d.GetOk("security_profile"); ok {
+		params.Properties.SecurityProfile = ExpandHDInsightSecurityProfile(v.([]interface{}))
+
+		params.Identity = &hdinsight.ClusterIdentity{
+			Type:                   hdinsight.ResourceIdentityTypeUserAssigned,
+			UserAssignedIdentities: make(map[string]*hdinsight.ClusterIdentityUserAssignedIdentitiesValue),
+		}
+
+		if params.Properties.SecurityProfile != nil && params.Properties.SecurityProfile.MsiResourceID != nil {
```

Comment:
`params.Properties.SecurityProfile` can still be nil here

Reply:
So I already added nil check for `if params.Properties.SecurityProfile != nil`

Reply:
Sorry my bad, misread this!

Reply:
Never mind. :)